aea/runtime.py

```diff
@@ -0,0 +1,238 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the implementation of runtime for economic agent (AEA)."""
+import asyncio
+import logging
+import threading
+from abc import ABC, abstractmethod
+from asyncio.events import AbstractEventLoop
+from enum import Enum
+from typing import Optional
+
+from aea.agent_loop import AsyncState
+from aea.helpers.async_utils import ensure_loop
+from aea.mail.base import AsyncMultiplexer
+
+if False:
+    # for mypy
+    from aea.agent import Agent
+
+
+logger = logging.getLogger(__name__)
+
+
+class RuntimeStates(Enum):
+    """Runtime states."""
+
+    initial = "not_started"
+    starting = "starting"
+    started = "started"
+    stopped = "stopped"
```

Comment:
what about "stopping"? Is there no use for it?

Reply:
will add and review all states and how it changed